app/uk/gov/hmrc/fileupload/microserviceGlobal.scala

```diff
@@ -167,13 +168,15 @@ object MicroserviceGlobal extends DefaultMicroserviceGlobal with RunMode {
   }
 
   lazy val testOnlyController = {
-    val fileRepository = uk.gov.hmrc.fileupload.read.file.Repository.apply(db)
-    val envelopeRepository = uk.gov.hmrc.fileupload.read.envelope.Repository.apply(db)
     val allEnvelopes = envelopeRepository.all _
+    val removeAllEnvelopes = () => envelopeRepository.removeAll()
+    val removeAllFiles = () => fileRepository.removeAll()
     implicit val reader = new UnitOfWorkReader(EventSerializer.toEventData)
     implicit val writer = new UnitOfWorkWriter(EventSerializer.fromEventData)
     val eventStore = new MongoEventStore(db)
```

Comment:
we should not create a new instance of readers, writers and mongoeventstore. We should reuse the eventStore instance we already have.